### PR TITLE
Decouple Data Fetch part in Topbeat

### PR DESCRIPTION
**Pull request by ruflin**
The data fetching and processing part was completely decoupled from Topbeat. This allows to reuse the functionality in metricbeat or other beats.

For the moment the variable functions stayed the same even though they could be simplified in a second step. The goal is to keep the changes as low as possible.

This PR does not change any functionality.

### CODE REVIEW

File: topbeat/system/process.go [hunk 1]
```diff
@@ -24,6 +25,12 @@ type Process struct {
 	Ctime    time.Time
 }
 
+type ProcStats struct {
+	ProcStats bool
```
Comment (andrewkroh) [inline topbeat/system/process.go]: Is there a better name for this? It has some stutter.

Comment (ruflin) [inline topbeat/system/process.go]: @andrewkroh There are several variables that now should be renamed with the restructuring. To keep the changes to a minimum I didn't rename any variables. If it is ok for you I would prefer to do this in a second PR.

File: topbeat/beater/topbeat.go [hunk 2]
```diff
@@ -49,9 +47,9 @@ func (tb *Topbeat) Config(b *beat.Beat) error {
 		tb.period = 10 * time.Second
 	}
 	if tb.TbConfig.Input.Procs != nil {
-		tb.procs = *tb.TbConfig.Input.Procs
+		tb.procStats.Procs = *tb.TbConfig.Input.Procs
```
Comment (monicasarbu) [inline topbeat/beater/topbeat.go]: I wouldn't place `procs` under `procStats` as I think it is a bit confusing.  `procStats` is just the configuration option that if enabled then statistics about the processes are exported, so I would see it more top level. I think `procStats` is not a good name and we should change it with something more obvious like `exportProcStats`.

File: topbeat/system/process.go [hunk 3]
```diff
@@ -149,3 +156,88 @@ func GetProcCpuPercentage(last *Process, current *Process) float64 {
...
+	for _, reg := range procStats.Procs {
+		matched, _ := regexp.MatchString(reg, name)
+		if matched {
+			return true
+		}
+	}
+	return false
+}
+
+func (procStats *ProcStats) InitProcStats() {
```
Comment (monicasarbu) [inline topbeat/system/process.go]: I would say it's enough to call it `Init`

**Comment by andrewkroh** [comment]
This is being merged because another PR is dependent upon it. The renames will occur in a follow-on pull request.

**Comment by ruflin** [comment]
@andrewkroh Thanks. Do make sure the second step does not get forgotton, I opened an issue here: https://github.com/elastic/beats/issues/1205

File: topbeat/beater/topbeat.go (hunk 2 above)
Comment (monicasarbu) [inline topbeat/beater/topbeat.go]: I suggest procStats.exportProcStats, procStats.procs and procStats.procMap

**Pull request merged**